### DISCUSSION TITLE
Should options.path be available by default?

### DISCUSSION THREAD

**Discussion by salmoro** [cyrilletuzi/vscode-angular-schematics#466] [Ideas]
Hi there and thanks for this neat extension.

I am authoring a schematics and I'm relying on the options.path which is available by default when running `ng g my-collection:my-schematics` from within a given directory (as mentioned in the angular documentation for schematics). Trying this extension seems that it always runs from the project root and therefore `path` is not supplied by schematics.

1. Is this the intended behavior?
2. If yes, how do you suggest I work around this?

Thanks!

**Comment by cyrilletuzi** [comment]
Hi @salmoro,

It's not easy to define what is the *intended* behavior for the extension, because as there is nearly no official documentation for the schematics API, the extension has been built *empirically* based on how the default Angular CLI schematics work.

In Angular CLI schematics:
- the first required argument (ng g component **path/to/component-name**) will automatically be set as the `name` option, which itself auto-manages the `path` option (set as invisible in the schematics, [example here](https://github.com/angular/angular-cli/blob/master/packages/schematics/angular/service/schema.json)),
- the `path` is indeed always relative to the workspace root.

So currently the extension works exactly the same way, but it does so not by hard-coding some options, but by following the schematics schema. It does filter options with:
- `$default": { "$source": "argv", "index": 0 }`
- `"visible": false`

So if your schematics does not set the `path` option as invisible or hidden, it should be proposed by the extension.

Is it possible to see your schematics code to check on a real example?

> as mentioned in the angular documentation for schematics

Also can you point me to this documentation?

**Comment by salmoro** [comment]
@cyrilletuzi Thanks for the quick reply.

Just looked at Angular's Schematics source code and I see that indeed they derive the path from the name option as you mentioned above.

Here's the mention in the docs:
>path: Overrides the path provided to the schematic. The default path value is based on the current working directory.

you can see it at the following link: https://angular.io/guide/schematics-for-libraries

BTW, the path is only available when using `ng generate ...` but not when using `schematics generate ...`

I don't have anything I can share right but if it'll be helpful I'll get you an example ASAP.

Thanks!

**Reply by cyrilletuzi** [reply]
Thanks for the information, waiting for your example.

**Comment by cyrilletuzi** [comment]
Hi @salmoro, for information, an extended version of the extension is now available: **Schematics Pro**. It has more advanced Angular features, but is also framework agnostic. Which means that it is no longer tied to the Angular CLI opacity and complexity. It just takes a few minutes to create custom schematics in Schematics Pro, with just a declarative configuration file (no headache logic code like in Angular CLI) and a templating syntax widely known and natively supported by VS Code (Handlebars).

You will find all the details on the [dedicated website](https://cyrilletuzi.gumroad.com/l/schematicspro/). Cheers.
